cabal-testsuite/cabal-testsuite.cabal

```diff
@@ -113,6 +114,6 @@ custom-setup
   -- we only depend on even stable releases of lib:Cabal
   -- and due to Custom complexity and ConstraintSetupCabalMaxVersion
   -- it can't be the current version
-  setup-depends: Cabal == 3.6.*,
-                 Cabal-syntax == 3.6.*,
+  setup-depends: Cabal == 3.8.*,
```

Comment:
Wow, so this works now? Could you remove the outdated comment, in that case?

Reply:
I will. This is weird but the version of cabal-install seems to make a difference here: I had to bump it (see a little above) to make this work. I can't explain why, but then again: my education in custom setups is severely lacking. What I deduced from numerous experiments is: it needs 3.8 here (in the custom setup of cabal-testsuite.cabal) and cabal-install 3.8 in the container. Anything else breaks on some jobs.

Reply:
A blind shot: are the disabled Windows tests broken because this is not cabal 3.6 now? 

I now remember I investigated it a bit and it was probably me who added the `ConstraintSetupCabalMaxVersion` keyword to be grepped next time somebody stumbles on it. But if 3.8 works, all the better and no need to investigate why. :)

Reply:
Oh! After grepping for `ConstraintSetupCabalMaxVersion`, I get where those mysterious constraints on `setup.Cabal` come from! I should say, it's a little suprising to me that it's the `cabal-install` version that affects this, but I don't want to argue for any change here. I think having a comment there is still helpful. We'd just need to update it a bit, e.g.:

>  -- we only depend on even stable releases of lib:Cabal
>  -- and due to Custom complexity and ConstraintSetupCabalMaxVersion
>  -- it has to be the current version plus you have to use the latest cabal-install release

How about that?

---

For Windows failures, I don't think they relate to cabal version: the error messages are clear that there's some issue with the way 9.4 uses the clang toolchain (which they migrated to in that release). I just did an experiment with `cabal build script.hs` (the source of most failures) using 9.4 and 3.6 on Windows, and they failed with the same error as 9.4/3.8 did (`windres` stuff). I'm afraid this is genuinely a GHC issue, and I am getting more convinced that there should be a bug report on their tracker.

Reply:
Fair enough. Please do as you say.